File: javaee3/src/main/java/com/ipartek/formacion/modelo/pojo/Alerta.java
package com.ipartek.formacion.modelo.pojo;

public class Alerta {
	
	public static final String TIPO_PRIMARY = "primary";
	public static final String TIPO_SUCCESS = "success";
	public static final String TIPO_DANGER = "danger";
	public static final String TIPO_WARNING = "warning";
	
	private String texto;
	private String tipo;
	
	public Alerta() {
		super();
		this.texto = "Error Inexperado";
		this.tipo = TIPO_DANGER;
	}

	public Alerta(String tipo, String texto) {
		this();
		this.texto = texto;
		this.tipo = tipo;
	}

	public String getTexto() {
		return texto;
	}

	public void setTexto(String texto) {
		this.texto = texto;
	}

	public String getTipo() {
		return tipo;
	}

	public void setTipo(String tipo) {
		this.tipo = tipo;
	}

	@Override
	public String toString() {
		return "Alerta [texto=" + texto + ", tipo=" + tipo + "]";
	}
	
}

File: dgt2/angular/src/app/model/alerta.ts
export class Alerta {

    public static TIPO_ERROR = 'danger';
    public static TIPO_SUCCESS = 'success';
    public static TIPO_INFO = 'info';
    public static TIPO_WARNING = 'warning';

    private _texto: string;
    private _tipo: string;

    constructor(texto: string, tipo = Alerta.TIPO_ERROR) {
        this._texto = texto;
        this._tipo = tipo;
    }

    public get texto(): string {
        return this._texto;
    }
    public set texto(value: string) {
        this._texto = value;
    }
    public get tipo(): string {
        return this._tipo;
    }
    public set tipo(value: string) {
        this._tipo = value;
    }

}
